src/ssl/support.cc

```diff
@@ -653,8 +657,8 @@ Ssl::Initialize(void)
 
     SQUID_OPENSSL_init_ssl();
 
-#if !defined(OPENSSL_NO_ENGINE)
     if (::Config.SSL.ssl_engine) {
+#if !defined(OPENSSL_NO_ENGINE) && OPENSSL_VERSION_MAJOR < 3
```

Comment:
Remove this version check to see why ```ssl_engine``` directive is deprecated.

```suggestion
#if !defined(OPENSSL_NO_ENGINE)
```